### PR TITLE
YLP-2160 patient list new design

### PR DESCRIPTION
**Pull request by fatb38**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

File: packages/yourloops/components/patient-list/custom-cells.tsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+  const onClickFlag = async (): Promise<void> => {
+    await flagPatient(patientId)
+  }
+
+  return (
+    <IconActionButton
+      icon={isFlagged
+        ? <FlagIcon
+          titleAccess="flag-icon-active"
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/custom-cells.tsx:73]: **💣 chore:** ‏`title` and `aria-label` must be human-readable strings (and translated) such as `Flag patient toto` or `Unflag patient toto`

File: packages/yourloops/components/patient-list/custom-cells.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  const { t } = useTranslation()
+
+  return (
+    <Tooltip
+      title={t(hasNewMessages ? 'unread-messages' : 'no-new-messages')}
+      aria-label={t(hasNewMessages ? 'unread-messages' : 'no-new-messages')}
+    >
+      <Box display="flex" justifyContent="center">
+        {hasNewMessages
+          ? <EmailIcon titleAccess="unread-messages-icon" color="primary" />
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/custom-cells.tsx:128]: **💣 chore:** ‏titleAccess should be human-readable ("The patient toto has send you new messages"). If this element has a title access, `EmailOpenIcon` should have one too

File: packages/yourloops/components/patient-list/patient-list.hook.tsx [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+  const navigate = useNavigate()
+  const gridApiRef = useGridApiRef()
+  const queryParams = useQueryParams()
+  const trNA = t('N/A')
+
+  const [selectedTab, setSelectedTab] = useState<PatientListTabs>(PatientListTabs.Current)
+  const [selectedFilter, setSelectedFilter] = useState<PatientListFilters>(PatientListFilters.All)
+  const [inputSearch, setInputSearch] = useState<string>('')
+  const [patientToRemoveForHcp, setPatientToRemoveForHcp] = useState<Patient | null>(null)
+  const [patientToRemoveForCaregiver, setPatientToRemoveForCaregiver] = useState<UserToRemove | null>(null)
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/patient-list.hook.tsx:94]: **❔ question:** ‏I think that you should have only a single state `patientToRemove`. Then the user role should be check when displaying which remove modal we should have. What do you think?

File: packages/yourloops/components/patient-list/patient-list.hook.tsx [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({ pageSize: 10, page: 0 })
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/patient-list.hook.tsx:95]: I don't think that this state should be here. It could be defined in the component. It is not used in the custom hook. :thinking: 

Same remark for `columnsVisibility`

File: packages/yourloops/components/patient-list/patient-list.hook.tsx [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+    })
+  }, [getPatientById, user])
+
+  const onCloseRemoveDialog = (): void => {
+    setPatientToRemoveForHcp(null)
+    setPatientToRemoveForCaregiver(null)
+  }
+
+  const onRowClick = (params: GridRowParams): void => {
+    navigate(`/patient/${params.id}`)
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/patient-list.hook.tsx:145]: **🎯 suggestion (non-blocking):** This might avoid a render in the router ‏
```suggestion
    navigate(`/patient/${params.id}/dashboard`)
```

File: packages/yourloops/components/patient-list/patient-list.hook.tsx [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+      {
+        type: 'actions',
+        field: PatientListColumns.Actions,
+        headerName: 'Actions',
+        renderCell: (params: GridRenderCellParams<GridRowModel, string>) => <ActionsCell patientId={params.value} onClickRemove={onClickRemovePatient} />
+      }
+    ]
+  }, [t, classes.mandatoryCellBorder, sortByUserName, selectedTab, getUserName, onClickRemovePatient])
+
+  const rows: GridRowsProp = useMemo(() => {
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/patient-list.hook.tsx:249]: **❔ question:** ‏So this is not a representation JSX of the rows, simply the props? Maybe rename it to `rowsProps`?

File: packages/yourloops/components/patient-list/patient-list.hook.tsx [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+    return filteredPatients.map((patient): GridRowModel => {
+      const { lastUpload } = getMedicalValues(patient.metadata.medicalData, trNA)
+      const toto = patient.alarms
```
Comment (TimCanu) [inline packages/yourloops/components/patient-list/patient-list.hook.tsx:252]:
```suggestion
      const alarms = patient.alarms
```

File: packages/yourloops/components/patient/add-dialog.tsx [hunk 8]
```diff
@@ -121,23 +119,24 @@ function AddDialog(props: AddDialogProps): JSX.Element {
...
-        key={team.id}
-        aria-label={team.name}
-      >
-        {team.name}
-      </MenuItem>
-    )
+  const addPatient = async (): Promise<void> => {
+    try {
+      setInProgress(true)
+      const team = teams.find(team => team.id === teamId)
```
Comment (TimCanu) [inline packages/yourloops/components/patient/add-dialog.tsx:125]: **🤓 nitpick:** ‏There is a `getTeam` method in the team hook

File: packages/yourloops/components/patient/add-dialog.tsx [hunk 9]
```diff
@@ -199,7 +199,17 @@ function AddDialog(props: AddDialogProps): JSX.Element {
               value={teamId}
               onChange={handleChangeTeam}
             >
-              {optionsTeamsElements}
+              <MenuItem id="patient-list-dialog-add-team-option-none" aria-label={t('none')} value="" key="none" />
```
Comment (TimCanu) [inline packages/yourloops/components/patient/add-dialog.tsx:202]: **❔ question:** ‏Is this long ass id really necessary? I don't see it being used

And what does `key="none"` do?

File: packages/yourloops/pages/home/home-page.tsx [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { type FunctionComponent } from 'react'
+import { useTheme } from '@mui/material/styles'
+import { GlobalStyles } from 'tss-react'
+import { PatientList } from '../../components/patient-list/patient-list'
+
+export const HomePage: FunctionComponent = () => {
```
Comment (TimCanu) [inline packages/yourloops/pages/home/home-page.tsx:33]: **💭 thought:** ‏This is only the home page of hcps and caregivers... It would be nice to find a better name someday

File: packages/yourloops/test/integration/pages/home-page/caregiver-home-page.spec.tsx [hunk 11]
```diff
@@ -142,14 +142,13 @@ describe('Caregiver home page', () => {
...
+    checkPatientList(UserRole.Caregiver)
 
-    const patientTableBody = within(screen.getByTestId('patient-table-body'))
+    const patientTableBody = within(screen.getByTestId('patient-list'))
     const patientData = patientTableBody.getByText(patientFullName)
     expect(patientData).toBeVisible()
 
-    const patientRow = screen.queryByTestId(`patient-row-${unmonitoredPatientAsTeamMember.userId}`)
-    const removePatientButton = within(patientRow).getByRole('button', { name: 'Remove patient unmonitored-patient@diabeloop.fr' })
+    const removePatientButton = screen.getByRole('button', { name: `Remove patient ${unmonitoredPatientAsTeamMember.userId}` })
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/caregiver-home-page.spec.tsx:151]: **❔ question:** ‏userId ? It would be way better to have the user email instead (as before)

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 12]
```diff
@@ -77,13 +77,12 @@ describe('HCP home page', () => {
...
     await checkHCPLayout(`${firstName} ${lastName}`, { teamName: teamThree.name }, AVAILABLE_TEAMS)
-    checkSecondaryBar(false, true)
+    checkPatientList()
 
     expect(screen.queryAllByLabelText('flag-icon-active')).toHaveLength(0)
     expect(screen.getAllByLabelText('flag-icon-inactive')).toHaveLength(4)
 
-    const patientRow = screen.queryByTestId(`patient-row-${unmonitoredPatient.userid}`)
-    const removeButton = within(patientRow).getByRole('button', { name: `Remove patient ${unmonitoredPatient.profile.email}`, hidden: true })
+    const removeButton = screen.getByRole('button', { name: `Remove patient ${unmonitoredPatient.userid}` })
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:85]: **❔ question:** ‏I feel like this test is missing the opportunity to test the full patient list. What about selecting it fully having a `toHaveTextContent`?

File: packages/yourloops/pages/home/home-page.tsx (hunk 10 above)
Comment (TimCanu) [reply packages/yourloops/pages/home/home-page.tsx:33]: It would sounds specific to HCPs and not caregiver. I don't have any idea for now. We can think of this later :slightly_smiling_face:

File: packages/yourloops/components/patient-list/patient-list.hook.tsx (hunk 3 above)
Comment (fatb38) [reply packages/yourloops/components/patient-list/patient-list.hook.tsx:94]: I decided to split this into two differents variables because each model needs different prop : a `Patient` for the HCP and `UserToRemove` for the Caregiver. I think it's better for readabilty and avoid some bugs to have two distincts states instead of a single that could be either of them.

File: packages/yourloops/pages/home/home-page.tsx (hunk 10 above)
Comment (fatb38) [reply packages/yourloops/pages/home/home-page.tsx:33]: `ProfessionalAndNonProfessionalMeansThatISNotAPatientHomePage` fits ?

File: packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx [hunk 13]
```diff
@@ -188,7 +188,7 @@ describe('HCP home page', () => {
 
     const select = within(addPatientDialog).getByTestId('patient-team-selector')
     fireEvent.mouseDown(within(select).getByRole('button'))
-    fireEvent.click(screen.getByRole('option', { name: teamTwo.name }))
+    fireEvent.click(screen.getByRole('option', { name: teamThree.name }))
```
Comment (TimCanu) [inline packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:191]: **❔ question:** ‏Why this change?

Comment (fatb38) [reply packages/yourloops/test/integration/pages/home-page/hcp-home-page.spec.tsx:191]: Why not ?

File: packages/yourloops/components/patient-list/patient-list.tsx [hunk 14]
```diff
@@ -0,0 +1,146 @@
...
+    [PatientListColumns.Patient]: true,
+    [PatientListColumns.TimeOutOfRange]: true,
+    [PatientListColumns.SevereHypoglycemia]: true,
+    [PatientListColumns.DataNotTransferred]: true,
+    [PatientListColumns.LastDataUpdate]: true,
+    [PatientListColumns.Messages]: user.isUserHcp() ?? false,
+    [PatientListColumns.Actions]: true
+  })
+
+  // const toggleColumnVisibility = (columnName: PatientListColumns): void => {
```
[the remark this answers is not in the document]
Comment (fatb38) [reply packages/yourloops/components/patient-list/patient-list.tsx:74]: added a todo, will be used with YLP-2154

**Comment by TimCanu** [comment]
You have some warnings in the ITs logs :wink:

**Review by TimCanu — APPROVED** [review]
